include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -108,19 +121,210 @@ __pattern_transform_reduce(_ExecutionPolicy&& __exec, _ForwardIterator __first,
     return __res;
 }
 
+template< bool _Inclusive>
+struct __single_group_scan
+{
+    template<typename _Policy, typename _InRng, typename _OutRng, typename _InitType, typename _BinaryOperation, typename _UnaryOp>
+    static void apply(_Policy const & __policy, _InRng __in, _OutRng __out, ::std::size_t __n, _InitType __init, const _BinaryOperation& __bin_op, const _UnaryOp& __unary_op, ::std::uint32_t __wg_size)
```

Comment:
Can we use `uint16_t` for `__wg_size`?

Reply:
> It may be good to come up with different names for the two `apply()` functions to describe what situations they apply to, or at least add some comments to that end.

I agree. I will name these something like `launch_static_bounds` and `launch_dynamic_bounds`.

> Can we use `uint16_t` for `__wg_size`?

Yes, will change.